### DISCUSSION TITLE
A clearer error message(s) during compilation procedure

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#248] [General]
I was writing a demo app for testing purposes on an idea I have and forgot to use `local` for a variable and got the following message:

```
tmp.nelua:1:1: from: AST node Block
do
^~
tmp.nelua:2:3: from: AST node Block
  x = 10
  ^~~~~~
tmp.nelua:2:3: error: undeclared symbol 'x'
  x = 10
  ^
```

About the first two messages that have to do with AST, is it something that the standard user should care about or not?

Also, shouldn't there be a helper output with the undeclared symbol to let user know he or she forgot to add `local` or `global`?

For a moment I was confused until I have realized my mistake.

Any thoughts on this @edubart ?

**Comment by edubart** [comment]
The first two AST messages gives you more context around what the error happened.

I noticed a few times users asking for improved compiler error messages, with more context information and suggestions. For now the current error messages is what we have, I have ideas to improve the parser and analyzer to make the compiler guide users mistakes more, but this will need to wait.

**Reply by stefanos82** [reply]
That's good to know.

Maybe this specific output could be under a different flag(?), something like `--include-ast-nodes-on-errors` or something like that?

Just food for thought.

No need to push yourself on this, I just wanted to know, that's all.
